platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -117,180 +118,198 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
     
     _loading = true;
     
-    dispatch_async(queue, ^{
-        _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
-            _loading = false;
+    __weak __typeof__(self) weakSelf = self;
+    _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
+        weakSelf.loading = false;
+        
+        
+        if (mbglError) {
+            NSString *description = @(mbgl::util::toString(mbglError).c_str());
+            NSDictionary *userInfo = @{NSLocalizedDescriptionKey: description};
+            NSError *error = [NSError errorWithDomain:MGLErrorDomain code:MGLErrorCodeSnapshotFailed userInfo:userInfo];
             
-            NSMutableArray *infos = [NSMutableArray array];
+            // Dispatch result to origin queue
+            dispatch_async(queue, ^{
+                _snapshotCallback = NULL;
+                completion(nil, error);
+            });
+        } else {
+            // This code is in place to test the _snapshotCallback release cycle.
+            dispatch_async(queue, ^{
```

Comment:
I use this code snippet to try to isolate GL's snapshotter generation. This way I detected that `_snapshotterCall` was not getting released. My hypothesis is that an object that is called before the lambda contains the code that is retaining the remaining memory.